src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/ManageController.cs

```diff
@@ -50,183 +51,191 @@ public async Task<IActionResult> Index()
-            StatusMessage = "Twoje hasło zostało zmienione";
+            await this.accountService.SignInAsync(user).ConfigureAwait(false);
+#pragma warning disable CA1303 // Do not pass literals as localized parameters
+            this.logger.LogInformation("User changed their password successfully.");
+#pragma warning restore CA1303 // Do not pass literals as localized parameters
+            this.StatusMessage = "Twoje hasło zostało zmienione";
 
-            return RedirectToAction(nameof(ChangePassword));
+            return this.RedirectToAction(nameof(this.ChangePassword));
         }
 
         [HttpGet]
         public async Task<IActionResult> SetPassword()
         {
-            var user = await _manageService.GetUserAsync(User);
+            var user = await this.manageService.GetUserAsync(this.User).ConfigureAwait(false);
             if (user == null)
             {
-                throw new ApplicationException($"Nie można załadować użytkownika z identyfikatorem '{_accountService.GetUserId(User)}'.");
+                throw new ArgumentException($"Nie można załadować użytkownika z identyfikatorem '{this.accountService.GetUserId(this.User)}'.");
             }
 
-            var hasPassword = await _manageService.HasPasswordAsync(user);
+            var hasPassword = await this.manageService.HasPasswordAsync(user).ConfigureAwait(false);
 
             if (hasPassword)
             {
-                return RedirectToAction(nameof(ChangePassword));
+                return this.RedirectToAction(nameof(this.ChangePassword));
             }
 
-            var model = new SetPasswordViewModel { StatusMessage = StatusMessage };
-            return View(model);
+            var model = new SetPasswordViewModel { StatusMessage = this.StatusMessage };
+            return this.View(model);
         }
 
         [HttpPost]
         [ValidateAntiForgeryToken]
         public async Task<IActionResult> SetPassword(SetPasswordViewModel model)
         {
-            if (!ModelState.IsValid)
+            if (!this.ModelState.IsValid)
             {
-                return View(model);
+                return this.View(model);
             }
 
-            var user = await _manageService.GetUserAsync(User);
+            var user = await this.manageService.GetUserAsync(this.User).ConfigureAwait(false);
             if (user == null)
             {
-                throw new ApplicationException($"Nie można załadować użytkownika z identyfikatorem '{_accountService.GetUserId(User)}'.");
+                throw new ArgumentException($"Nie można załadować użytkownika z identyfikatorem '{this.accountService.GetUserId(this.User)}'.");
             }
 
-            var addPasswordResult = await _manageService.AddPasswordAsync(user, model.NewPassword);
+#pragma warning disable CA1062 // Validate arguments of public methods
+            var addPasswordResult = await this.manageService.AddPasswordAsync(user, model.NewPassword).ConfigureAwait(false);
```

Comment:
in the first line of this method, check if SetPasswordVieModel is not null (if null, throw ArgumentNullException) and remove pragma warning